lib/Chat/Parser/SystemMessage.php

```diff
@@ -763,18 +763,15 @@ protected function getFileFromShare(Room $room, ?Participant $participant, strin
 				}
 
 				$fullPath = $node->getPath();
-				$pathSegments = explode('/', $fullPath, 4);
-				$name = $node->getName();
-				$size = $node->getSize();
-				$path = $pathSegments[3] ?? $name;
+				$userFolder->getRelativePath($fullPath);
 			} else {
 				$node = $share->getNode();
 				$name = $node->getName();
 				$size = $node->getSize();
 
 				$fullPath = $node->getPath();
-				$pathSegments = explode('/', $fullPath, 4);
-				$path = $pathSegments[3] ?? $name;
+				$userFolder = $this->rootFolder->getUserFolder($participant->getAttendee()->getActorId());
+				$userFolder->getRelativePath($fullPath);
```

Comment:
Just added it here or should I only have one instance of $userFolder, initialized after the first if statement